mlflow/utils/rest_utils.py

```diff
@@ -169,6 +171,54 @@ def call_endpoint(host_creds, endpoint, method, json_body, response_proto):
     return response_proto
 
 
+# Response codes that generally indicate transient network failures and merit client retries,
+# based on guidance from cloud service providers
+# (https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines)
+TRANSIENT_FAILURE_RESPONSE_CODES = [
+    408, # Request Timeout
+    429, # Too Many Requests
+    500, # Internal Server Error
+    502, # Bad Gateway
+    503, # Service Unavailable
+    504, # Gateway Timeout
+]
+
+
+def cloud_storage_http_request(method, *args, **kwargs):
+    """
+    Performs an HTTP PUT/GET request using Python's `requests` module with an automatic retry
+    policy of `retry_attempts` using exponential backoff for the following response codes:
+
+        - 408 (Request Timeout)
+        - 429 (Too Many Requests)
+        - 500 (Internal Server Error)
+        - 502 (Bad Gateway)
+        - 503 (Service Unavailable)
+        - 504 (Gateway Timeout)
+
+    :method: string of 'PUT' or 'GET', specify to do http PUT or GET
+    :args: Positional arguments to pass to `requests.Session.put/get()`
+    :kwargs: Keyword arguments to pass to `requests.Session.put/get()`
+    """
+    retry_attempts = kwargs.get('retry_attempts', 5)
+    retry_strategy = Retry(
+        total=None,
+        status=retry_attempts,
+        status_forcelist=TRANSIENT_FAILURE_RESPONSE_CODES,
+        backoff_factor=1,
+    )
```

Comment:
To address https://github.com/mlflow/mlflow/pull/4044/files#r570841358, I think we can make the following change:

```suggestion
    retry_strategy = Retry(
        total=None,
        connect=0, # Don't retry on connect-related errors raised before a request reaches a remote server
        read=1, # Retry once for errors reading the response from a remote server,
        redirect=3, # Limit the number of redirects to avoid infinite redirect loops
        status=retry_attempts, # Retry a specified number of times for response codes indicating transient failures, 
        other = 0, # Don't retry for other, unclassified errors 
        status_forcelist=TRANSIENT_FAILURE_RESPONSE_CODES,
        backoff_factor=1,
    )
```